platform/engineering/design-docs/2021-01-13-web-component-library.md

```diff
@@ -0,0 +1,139 @@
+# Web Component library
+
+**Author(s):** Brooks Johnson  
+**Last Updated:** January 13, 2021  
+**Status:** **Draft** | In Review | Approved  
+**Approvers:** _Chris Valarida_ \[ \], _Michael Fleet_ \[ \]
+
+
+## Overview
+
+### Objective
+
+The goal is to establish a component library based on Web Components, which are a browser standard. This will supercede the current `formation-react` library and some of the liquid HTML templates currently in `vets-website`. A non-goal is creating an entirely new visual experience as part of the component library.
+
+The intended audience for this document is frontend engineers.
+
+### Background
+
+We currently have some "components" implemented in HTML/CSS (and sometimes JS) in the `formation` npm package. These can be used anywhere where `formation` is loaded. We also have `formation-react`, which is a package with some of those patterns implemented as React components. These can only be used in React apps. Lastly, in `vets-website` we have some Liquid templates which implement some of the components/patterns of the design system, and these are only used for content builds.
+
+Maintaining multiple sources of truth is a burden for all teams involved and leads to wasted time.
+
+### High Level Design
+
+The goal is to establish a single codebase and documentation for the implementation of design system components. This will allow the components to be used in React apps, static HTML, or any other web framework.
+
+The Web Component library will live in its own repo and will be added to `vets-website` as a dependency, where the components will be loaded and defined for the browser to use. As more components are added to the Web Component version of the library, the older React-specific components will be phased out.
+
+
+## Specifics
+
+### Detailed Design
+
+In the `component-library` repo, we will start a fresh orphan branch for the Web Component version of the library to live. If we reach a point where the React version of the library is no longer in use, we will rebase the orphan Web Component branch off of master (in order to cleanly preserve the history) and that will become the new default branch going forward.
```

Comment:
Have you tried doing this to make sure it'll work like you want? I'm not super familiar with rebasing.

When you say "orphan branch," the web component branch will share a history with `master`, right?

Reply:
I left out some detail here. I have done all of this locally except for the rebasing, but I'm pretty confident it will work like I expect. I created a branch off of `master` called `wiped` where I deleted everything but the `.git` directory, and then I created the orphan branch off of that. That gave me a branch with a clean slate in the same repo, but none of the history.

---

I just did the rebase as a sanity check, and the only conflicts were `.gitignore`, `package.json`, and the `.storybook` `main.js` and `preview.js` files, which is what I expected.

The result is [this branch](https://github.com/department-of-veterans-affairs/component-library/tree/wc-pilot-rebase-test). The commits are clean, but I have some more things to delete which isn't a big deal.